src/commonTest/kotlin/operations/logic/IfTest.kt

```diff
@@ -1,235 +1,223 @@
+        ),
+        TestInput(
+            expression = mapOf("if" to listOf(true)),
+            resultValue = true
+        ),
+        TestInput(
+            expression = mapOf("if" to listOf(false)),
+            resultValue = false
+        ),
+        TestInput(
+            expression = mapOf("if" to listOf("apple")),
+            resultValue = "apple"
+        ),
+        TestInput(
+            expression = mapOf("if" to listOf(true, "apple")),
+            resultValue = "apple"
+        ),
+        TestInput(
+            expression = mapOf("if" to listOf(emptyList<Any>(), "apple", "banana")),
+            resultValue = "banana"
+        ),
+        TestInput(
+            expression = mapOf("if" to listOf(listOf(1), "apple", "banana")),
+            resultValue = "apple"
+        ),
+        TestInput(
+            expression = mapOf("if" to listOf(listOf(1, 2, 3, 4), "apple", "banana")),
+            resultValue = "apple"
+        ),
+        TestInput(
+            expression = mapOf("if" to listOf("", "apple", "banana")),
+            resultValue = "banana"
+        ),
+        TestInput(
+            expression = mapOf("if" to listOf("zucchini", "apple", "banana")),
+            resultValue = "apple"
+        ),
+        TestInput(
+            expression = mapOf("if" to listOf("0", "apple", "banana")),
+            resultValue = "apple"
+        ),
+        TestInput(
+            expression = mapOf("if" to listOf(mapOf("+" to "0"), "apple", "banana")),
+            resultValue = "banana"
+        ),
+        TestInput(
+            expression = mapOf("if" to listOf(mapOf("+" to "1"), "apple", "banana")),
+            resultValue = "apple"
+        ),
+        TestInput(
+            expression = mapOf("if" to listOf(0, "apple", "banana")),
+            resultValue = "banana"
+        ),
+        TestInput(
+            expression = mapOf("if" to listOf(1, "apple", "banana")),
+            resultValue = "apple"
+        ),
+        TestInput(
+            expression = mapOf("if" to listOf(3.1416, "apple", "banana")),
+            resultValue = "apple"
+        ),
+        TestInput(
+            expression = mapOf("if" to listOf(-1, "apple", "banana")),
+            resultValue = "apple"
+        ),
+        TestInput(
+            expression = mapOf("if" to listOf(true, "yes", "no")),
+            resultValue = "yes"
+        ),
+        TestInput(
+            expression = mapOf("if" to listOf(false, "yes", "no")),
+            resultValue = "no"
+        ),
+        TestInput(
+            expression = mapOf(
+                "if" to listOf(
+                    mapOf("<" to listOf(mapOf("var" to "temp"), 0)), "freezing",
+                    mapOf("<" to listOf(mapOf("var" to "temp"), 100)), "liquid",
+                    "gas"
+                )
+            ),
+            data = mapOf("temp" to 55),
+            resultValue = "liquid"
+        ),
+    ),
+    failureResultTestInput = listOf(
```

Comment:
[not a blocker]
For this type of test input resultValue is not used - so probably other data type should be used here?